orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/AuthenticationAware.kt

```diff
@@ -28,10 +28,11 @@ interface AuthenticationAware {
   val stageNavigator: StageNavigator
 
   fun StageExecution.withAuth(block: () -> Unit) {
-    val authenticatedUser = stageNavigator
+    val firstAncestorOrNull = stageNavigator
```

Comment:
Curious about the motivation here if nothing else uses `firstAncestorOrNull` except to figure out `authenticatedUser`.

Reply:
I think was part of the strategy to implement an AuthenticatedStage as interface, this interface allows an implementing StageDefinitionBuilder to override the default pipeline authentication context. In this case, was retrieving the authenticated user from the previous stage, but when a stage is skipped the chain of authenticated user permissions  is broken.

Reply:
I think I'm asking a simpler question...like, there's a diff here to add a new variable, but there's nothing else that uses the new variable...so, do we need the new variable after all?

Reply:
Oh sorry my error, I did not understand, now I know about your recommendation. Thank you, I added a simplification to prevent that "one-call variables"

Reply:
I was actually thinking of something different, to minimize the diffs, but I like splitting it out to a separate method too.